### PR TITLE
Added test for user being signed up with share_request_key

### PR DESCRIPTION
**Pull request by ijunaidfarooq**
/cc @azharmalik3

just as an information to others..

Doing sign up with a share key and without it doesn't make any difference but only in one place. 
e.g when a user is being signed up with a share key then we add its confirmed at, in this case from evercam media. But if its not with share key then there will be no confirmed at. (It will be added from dashboard on click the link which being sent in the mail.)So in test i more specifically getting 200 status along with checking that.. confirmed_at is present or not.

### CODE REVIEW

**Comment by marcoherbst** [comment]
@ijunaidfarooq , the real difference is that a sharekey will also give the new user access to a shared camera and will not add the demo camera, correct?

**Comment by ijunaidfarooq** [comment]
Yeah right.. but we are discussing user sign up.. not its camera or access
to camera.. user signing up will give status 200 in both cases.. but the
main difference is.. the returned user will have a confirmed at date..in
case of share request key..

**Comment by marcoherbst** [comment]
Ok, I get it. ConfirmedAt = Email Address is confirmed. Makes sense.

**Comment by ijunaidfarooq** [comment]
Yeah.. the main point is user already got the shared key link in email.. so
its coming from email itself.. so we are confirming it in this case
already. So

Confirmed at.. is email authenticated..

File: test/controllers/user_controller_test.exs [hunk 1]
```diff
@@ -61,11 +63,28 @@ defmodule EvercamMedia.UserControllerTest do
...
     response =
       build_conn()
       |> post("/v1/users", params)
 
+    signed_up_user =
+      response.resp_body
+      |> Poison.decode!
+      |> Map.get("users")
+      |> List.first
+      |> Enum.find(fn {key, _val} -> key == "confirmed_at" end)
```
Comment (azharmalik3) [inline test/controllers/user_controller_test.exs]: Remove `Enum.fine...` and `elem(1)`. We don't need these two lines.

File: test/controllers/user_controller_test.exs [hunk 2]
```diff
@@ -61,11 +63,28 @@ defmodule EvercamMedia.UserControllerTest do
...
+      |> elem(1)
+
     assert response.status == 200
+    assert signed_up_user != nil
```
Comment (azharmalik3) [inline test/controllers/user_controller_test.exs]: check confirmed_at like this `signed_up_user["confirmed_at"] != nil`

File: test/controllers/user_controller_test.exs (hunk 1 above)
Comment (ijunaidfarooq) [inline test/controllers/user_controller_test.exs]: Done!

File: test/controllers/user_controller_test.exs (hunk 2 above)
Comment (ijunaidfarooq) [inline test/controllers/user_controller_test.exs]: done!

File: test/controllers/user_controller_test.exs (hunk 1 above)
Comment (ijunaidfarooq) [inline test/controllers/user_controller_test.exs]: I have made changes to other PR as well.. https://github.com/evercam/evercam-media/pull/472/files

**Pull request merged**